packages/models/src/store/store.ts

```diff
@@ -10,57 +12,129 @@ import type {
+      'offset' in option &&
+      typeof option.offset === 'number' &&
+      'length' in option &&
+      typeof option.length === 'number'
+    ) {
+      return [option.offset || 0, option.length || 0]
+    }
+    if ('offset' in option && typeof option.offset === 'number') {
+      return [option.offset || 0, 0]
+    }
+    if ('length' in option && typeof option.length === 'number') {
+      return [0, option.length || 0]
+    }
+    return [0, 0]
+  }
+
+  private deserializeScript<T extends 'lock' | 'type'>(
+    scriptValue: Script,
+    scriptOption: unknown,
+    type: T,
+  ): GetFullStorageStruct<StructSchema>[T] {
+    if (typeof scriptOption !== 'object' || scriptOption === null) throw new Error()
+    const res = {} as GetFullStorageStruct<StructSchema>[T]
+    if ('args' in scriptOption) {
+      this.assetStorage(this.getStorage([type, 'args']))
+      const [offset, length] = this.getOffsetAndLength(scriptOption.args)
+      res['args'] = this.getStorage([type, 'args'])?.deserialize(getUint8ArrayfromHex(scriptValue.args, offset, length))
     }
+    if ('codeHash' in scriptOption) {
+      this.assetStorage(this.getStorage([type, 'codeHash']))
+      const [offset, length] = this.getOffsetAndLength(scriptOption.codeHash)
+      res['codeHash'] = this.getStorage([type, 'codeHash'])?.deserialize(
+        getUint8ArrayfromHex(scriptValue.codeHash, offset, length),
+      )
+    }
+    return res
+  }
+
+  private deserializeCell({ cell, witness }: UpdateStorageValue): GetStorageStruct<StructSchema> {
+    const res: Record<string, unknown> = {}
+    if ('data' in this.schemaOption) {
+      this.assetStorage(this.getStorage('data'))
+      const [offset, length] = this.getOffsetAndLength(this.schemaOption.data)
+      res.data = this.getStorage('data')?.deserialize(getUint8ArrayfromHex(cell.data, offset, length))
+    }
+    if ('witness' in this.schemaOption) {
+      this.assetStorage(this.getStorage('witness'))
+      const [offset, length] = this.getOffsetAndLength(this.schemaOption.witness)
+      res.witness = this.getStorage('witness')?.deserialize(getUint8ArrayfromHex(witness, offset, length))
+    }
+    if ('lock' in this.schemaOption) {
+      res.lock = this.deserializeScript<'lock'>(cell.cellOutput.lock, this.schemaOption.lock, 'lock')
+    }
+    if ('type' in this.schemaOption && cell.cellOutput.type) {
+      res.type = this.deserializeScript<'type'>(cell.cellOutput.type, this.schemaOption.type, 'type')
+    }
+    return res as GetStorageStruct<StructSchema>
   }
 
-  private addState(addStates: Record<OutPointString, GetStorageStruct<StructSchema>>) {
-    this.states = {
-      ...this.states,
-      ...addStates,
+  private addState({ cell, witness }: UpdateStorageValue) {
+    if (cell.outPoint) {
+      const outpoint = `${cell.outPoint?.txHash}${cell.outPoint.index.slice(2)}`
+      this.states[outpoint] = this.deserializeCell({ cell, witness })
+      this.chainData[outpoint] = { cell, witness }
     }
```

Comment:
Expect to throw if `cell` doesn't have its out point

Reply:
I thought no added is enough.